daemon/logger/rollover/rollover.go

```diff
@@ -0,0 +1,144 @@
+package rollover
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/docker/docker/daemon/logger"
+	"github.com/docker/docker/pkg/jsonlog"
+)
+
+// RolloverLogger is Logger implementation for rollover docker logging:
+// JSON objects to file
+type RolloverLogger struct {
+	buf      *bytes.Buffer
+	f        *os.File // store for closing
+	capacity int64
+	n        int
+}
+
+func New(filename string, capacity string, n int) (logger.Logger, error) {
+	var capVal int64
+	if capacity != "" {
+		var multiplier int64
+		capString := capacity[:len(capacity)-1]
```

Comment:
just lowercase or uppercase the rune so you don't have to have so many case statements